### PR TITLE
CNV-19844: Adding utilization charts to vm metrics tab

### PR DESCRIPTION
**Pull request by metalice**
Signed-off-by: Matan Schatzman <mschatzm@redhat.com>

## 📝 Description

Moving charts to shared locations.
Add a responsive hook, so charts resize will be dynamically
Add charts to the metrics tab

## 🎥 Demo

image

### CODE REVIEW

File: src/utils/components/Charts/hooks/useResponsiveCharts.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        setWidth(node.getBoundingClientRect().width);
+        setHeight(node.getBoundingClientRect().height);
+      };
+      sizeSetter();
+      setListener(getResizeObserver(containerRef?.current, sizeSetter));
+      containerRef.current = node;
+    }
+  }, []);
+
+  useEffect(() => () => listener && listener(), [listener]);
```
Comment (glekner) [inline src/utils/components/Charts/hooks/useResponsiveCharts.tsx:27]: can you make this effect more readable with blocks?.
also, don't you need a cleanup here?

File: src/views/virtualmachines/details/tabs/metrics/virtual-machine-metrics-tab.scss [hunk 2]
```diff
@@ -40,3 +40,7 @@
     }
   }
 }
+.util-threshold-chart {
+  width: 100%;
+  height: 150px;
```
Comment (glekner) [inline src/views/virtualmachines/details/tabs/metrics/virtual-machine-metrics-tab.scss:45]: why hardcoded height?
can't we use flex here?

File: src/utils/components/Charts/hooks/useResponsiveCharts.tsx (hunk 1 above)
Comment (metalice) [reply src/utils/components/Charts/hooks/useResponsiveCharts.tsx]: cool 😎

**Pull request merged**